src/components/ProjectCard/ProjectHoverCard.tsx

```diff
@@ -24,9 +26,37 @@ export const ProjectHoverCard: FC<ProjectCardProps> = ({
   const router = useRouter();
   const { address } = useAccount();
   const [isModalOpen, setModalOpen] = useState(false);
+  const [totalAmount, setTotalAmount] = useState<number>(0);
+  const [tokenPrice, setTokenPrice] = useState(1);
 
   const openModal = () => setModalOpen(true);
   const closeModal = () => setModalOpen(false);
+  useEffect(() => {
+    const fetchPrice = async () => {
+      const price = await fetchTokenPrice('wmatic');
+      setTokenPrice(price);
+    };
+
+    fetchPrice();
+  }, []);
```

Comment:
Thanks, @lovelgeorge99!
Fetching the "wamtic" token price for each project card individually isn't the most efficient approach. I recommend using react-query with caching configurations to reduce overhead and improve performance.